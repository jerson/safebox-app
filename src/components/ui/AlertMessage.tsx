import React, {useEffect} from 'react';
import {StyleSheet, TextStyle, View, ViewStyle} from 'react-native';
import Touchable, {TouchableProps} from './Touchable';
import Colors from '../../modules/constants/Colors';
import Text from './Text';
import Icon from 'react-native-vector-icons/Feather';
import useAnimatedState from '../hooks/useAnimatedState';

const styles = StyleSheet.create({
  container: {
    padding: 12,
    borderRadius: 10,
    backgroundColor: Colors.danger,
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row',
  } as ViewStyle,
  text: {
    color: Colors.white,
    fontSize: 13,
  } as TextStyle,
  icon: {
    fontSize: 18,
    marginRight: 6,
    marginTop: 2,
    alignSelf: 'flex-start',
    color: Colors.white,
    opacity: 0.8,
  } as TextStyle,
  content: {
    flex: 1,
  } as ViewStyle,
});

export interface AlertMessageProps extends TouchableProps {
  message?: string;
  timeout?: number;
  icon?: string;
  color?: string;
  onTimeout?: () => void;
}

function AlertMessage(props: AlertMessageProps) {
  const {
    style,
    onPress,
    icon,
    color,
    timeout,
    onTimeout,
    message,
    ...extraProps
  } = props;

  const [visible, setVisible] = useAnimatedState(true);

  useEffect(() => {
    if ((timeout || 0) < 1) {
      return;
    }
    const listener = setTimeout(() => {
      setVisible(false);
      typeof onTimeout === 'function' && onTimeout();
    }, timeout);

    return () => {
      clearTimeout(listener);
    };
  }, [timeout]);

  if (!visible) {
    return null;
  }
  return (
    <Touchable
      style={[styles.container, {backgroundColor: color}, style]}
      onPress={onPress}
      {...extraProps}>
      <Icon style={styles.icon} name={icon || 'alert-circle'} />
      <View style={styles.content}>
        <Text style={styles.text}>{message}</Text>
      </View>
    </Touchable>
  );
}
AlertMessage.defaultProps = {
  timeout: 5000,
  icon: 'alert-circle',
  color: Colors.danger,
};
export default AlertMessage;
