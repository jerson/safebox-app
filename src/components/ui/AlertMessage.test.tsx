import React from 'react';
import * as renderer from 'react-test-renderer';
import AlertMessage, { AlertMessageProps } from './AlertMessage';

const props: AlertMessageProps = {
  message: 'sample'
};
test('renders without crashing', () => {
  const snapshot = renderer.create(<AlertMessage {...props} />).toJSON();
  expect(snapshot).toBeTruthy();
});

test('render just views', () => {
  const snapshot = renderer.create(<AlertMessage {...props} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
