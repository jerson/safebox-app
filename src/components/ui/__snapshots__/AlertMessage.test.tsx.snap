// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render just views 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#e91e63",
        "borderRadius": 10,
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 12,
      },
      Object {
        "backgroundColor": "#e91e63",
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Object {
          "alignSelf": "flex-start",
          "color": "#FFFFFF",
          "fontSize": 18,
          "marginRight": 6,
          "marginTop": 2,
          "opacity": 0.8,
        },
        Object {
          "fontFamily": "Feather",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ï„‚
  </Text>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Nunito",
            "fontSize": undefined,
            "fontStyle": "normal",
            "fontWeight": "400",
          },
          Object {},
          Object {
            "color": "#FFFFFF",
            "fontSize": 13,
          },
        ]
      }
    >
      sample
    </Text>
  </View>
</View>
`;
